File: src/dsa/meeting_one_submission/Capistrano.java
import java.util.Scanner;

public class Capistrano {
    public static void main(String[] args) {
        // Sorted list of names
        String[] names = { "Anna", "Ben", "Carlos", "Diana", "Ella", "Francis", "Grace", "Hannah" };

        Scanner sc = new Scanner(System.in);
        System.out.print("Enter name to search: ");
        String target = sc.nextLine();

        int left = 0;
        int right = names.length - 1;
        boolean found = false;

        while (left <= right) {
            int mid = (left + right) / 2;

            // Print the left, mid, and right values at each step
            System.out.println("\nLeft: " + left + " (" + names[left] + ")");
            System.out.println("Mid: " + mid + " (" + names[mid] + ")");
            System.out.println("Right: " + right + " (" + names[right] + ")");

            int result = target.compareToIgnoreCase(names[mid]);

            if (result == 0) {
                System.out.println("\n✅ " + target + " found at index " + mid + "!");
                found = true;
                break;
            } else if (result < 0) {
                System.out.println(target + " comes before " + names[mid] + " → move left");
                right = mid - 1;
            } else {
                System.out.println(target + " comes after " + names[mid] + " → move right");
                left = mid + 1;
            }
        }

        if (!found) {
            System.out.println("\n❌ " + target + " not found in the list.");
        }

        sc.close();
    }
}

File: src/dsa/meeting_one_submission/capistrano.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1
    print(f"Searching for: {target}")

    while left <= right:
        print(f"Left: {left}  Right: {right}")
        mid = (left + right) // 2
        print(f"Checking middle index: {mid}  Value: {arr[mid]}")

        if arr[mid] == target:
            print(f"Found {target} at index {mid}")
            return mid
        elif arr[mid] < target:
            print(f"{arr[mid]} is before {target} -> moving right")
            left = mid + 1
        else:
            print(f"{arr[mid]} is after {target} -> moving left")
            right = mid - 1

    print(f"{target} not found in the list.")
    return -1

words = ["1", "2", "3", "4", "5", "6", "7"]
binary_search(words, "2")